port/zos390/omrvmem.c

```diff
@@ -510,16 +510,24 @@ reserve1MPageableBelowBar(struct OMRPortLibrary *portLibrary, J9PortVmemIdentifi
  * @return NULL on failure, otherwise pointer to the memory that was allocated.
  */
 static void *
-reservePagesAboveBar(struct OMRPortLibrary *portLibrary, J9PortVmemIdentifier *identifier, uintptr_t mode, uintptr_t byteAmount, uintptr_t pageSize, uintptr_t pageFlags, uintptr_t options, OMRMemCategory *category)
+reservePagesAboveBar(struct OMRPortLibrary *portLibrary, J9PortVmemIdentifier *identifier, struct J9PortVmemParams *params, OMRMemCategory *category)
 {
 	/* The maximum amount of memory that can be allocated with userExtendedPrivateAreaMemoryType detected for 2To32G and 2To64 */
 #define MAX_2TO32G_MEMORY_IN_MB ((UDATA)30*1024)	/* unit is MB i.e. 30*1024 MB = 30 GB */
 #define MAX_2TO64G_MEMORY_IN_MB ((UDATA)62*1024)	/* unit is MB i.e. 62*1024 MB = 62 GB */
 
-	void *ptr = NULL;
+    uintptr_t mode       = params->mode;
+    uintptr_t byteAmount = params->byteAmount;
+    uintptr_t pageSize   = params->pageSize;
+    uintptr_t pageFlags  = params->pageFlags;
+    uintptr_t options    = params->options;
+    void* startAddress   = (void*)params->startAddress;
```

Comment:
As prototype to test new feature is ok, but please do not forget that in real life `params->startAddress` can be NULL (means any location is ok and it is default initial value).

Reply:
In this case, I'll need to make two types of `IARV64 REQUEST=GETSTOR,....` calls (in `IARV64.c` depending on whether this is NULL.